libraries/chain/controller.cpp

```diff
@@ -691,10 +698,14 @@ struct building_block {
                   .new_protocol_feature_activations = new_protocol_feature_activations()
                };
 
+               // get current block reference
+               block_ref current_block {parent_id(), timestamp()};
```

Comment:
This doesn't look correct. How can current block have parent id and current timestamp?

Reply:
Thank you Kevin for catching this! I have fixed it and also renamed it to parent_block to reduce confusion.